bctw-api/src/import/csv.ts

```diff
@@ -233,26 +247,229 @@ const importXlsx = async function (req: Request, res: Response): Promise<void> {
   await parseXlsx(file, id, onFinishedParsing);
 };
 
+
+
+
+const createNewLocationFromRow = (bctw_animal: IAnimalDeviceMetadata) => {
+  if(bctw_animal.capture_longitude || bctw_animal.capture_latitude) {
+    const locationBody = {
+      location_id: uuidv4(),
+      longitude: bctw_animal.capture_longitude,
+      latitude: bctw_animal.capture_latitude
+    }
+    return locationBody;
+  }
+  if(bctw_animal.capture_utm_easting || bctw_animal.capture_utm_northing) {
+    const [longitude, latitude] = projectUTMToLatLon( bctw_animal.capture_utm_northing, bctw_animal.capture_utm_easting, bctw_animal.capture_utm_zone ?? 10);
+    const locationBody = {
+      location_id: uuidv4(),
+      longitude: longitude,
+      latitude: latitude
+    }
+    return locationBody;
+  }
+  return null;
+}
+
+const createNewCaptureFromRow = (critter_id: string, bctw_animal: IAnimalDeviceMetadata, location_id: string | null) => {
+  const captureBody = {
+    capture_id: uuidv4(),
+    critter_id: critter_id,
+    capture_location_id: location_id,
+    capture_timestamp: dayjs(bctw_animal.capture_date).format(),
+    capture_comment: bctw_animal.capture_comment
+  } 
+
+  return captureBody;
+}
+
+const createNewMarkingsFromRow = (critter_id: string, capture_id: string, bctw_animal: IAnimalDeviceMetadata) => {
+  const bctw_markings = getCritterbaseMarkingsFromRow(bctw_animal);
+  const critterbase_markings: any[] = [];
+  if(bctw_markings.length) {
+    for(const m of bctw_markings) {
+      m.critter_id = critter_id,
+      m.capture_id = capture_id;
+      critterbase_markings.push(m);
+    }
+  }
+  return critterbase_markings;
+}
+
+const createNewMortalityFromRow = (critter_id: string, bctw_animal: IAnimalDeviceMetadata) => {
+  if(bctw_animal.mortality_date) {
+    const mortalityBody = {
+      critter_id: critter_id,
+      mortality_timestamp: dayjs(bctw_animal.mortality_date).format(),
+      mortality_comment: bctw_animal.mortality_comment
+    }
+    return mortalityBody;
+  }
+  return null;
+}
+
+interface IBulkCritterbasePayload {
+  critters: Record<string, any>[],
+  markings: Record<string, any>[],
+  collections: Record<string, any>[]
+  locations: Record<string, any>[],
+  captures: Record<string, any>[],
+  mortalities: Record<string, any>[]
+}
+
+const insertTemplateAnimalIntoCritterbase = async (bctw_animal: any, bulk_payload: IBulkCritterbasePayload) => {
+  
+  const new_critter_id: string = uuidv4();
+
+  const critterBody = {
+    critter_id: new_critter_id,
+    wlh_id: bctw_animal.wlh_id,
+    animal_id: bctw_animal.animal_id,
+    sex: bctw_animal.sex,
+    taxon_name_common: bctw_animal.species
```

Comment:
This may need a lookup to set the reference taxon_id, like what you did with collection_unit and markings (ignore this if I missed it somewhere).